src/Avalonia.Base/Utilities/StringTokenizer.cs

```diff
@@ -8,11 +8,11 @@ namespace Avalonia.Utilities
 #if !BUILDTASK
     public
 #endif
-    record struct StringTokenizer : IDisposable
+    ref struct StringTokenizer
```

Comment:
I believe that's a breaking change, so we can't have it until 12.0.

Not sure why is this class public in the first place.

We can probably a separate internal struct though.

Reply:
I could add an `internal ref struct SpanStringTokenizer` which will effectively be just a copy of the normal `StringTokenizer` but with the `ReadOnlySpan<char>` changes. We could then update all references in Avalonia itself while also keeping `StringTokenizer` as it is right now until we version bump to 12.0. Would that be an acceptable solution?

Reply:
I think so, yes. Also mark the current one as [Obsolete].

Reply:
We should also add a comment  "TODO12" and probably add it also here https://github.com/AvaloniaUI/Avalonia/issues/12276

This in order to increase the chance it doesn't get forgotten.